packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -2930,9 +2956,17 @@ class _InputDateRangePickerState extends State<_InputDateRangePicker> {
     return startError == null && endError == null;
   }
 
-  DateTime? _parseDate(String? text) {
+  DateTime? _parseDate(String? text, intl.DateFormat? dateFormat) {
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
-    return localizations.parseCompactDate(text);
+    DateTime? date;
+    if(dateFormat != null){
+      try{
+        date = dateFormat.parse(text!);
+      } on FormatException {
+        date = null;
+      }
+    }
+    return dateFormat == null ? localizations.parseCompactDate(text) : date;
```

Comment:
`localizations` property is being used in one place only, we can inline it in the place of usage:

`MaterialLocalizations.of(context).parseCompactDate(text)`

Reply:
Yes that makes sense, I have changed this now.